lib/auth/result.rb

```diff
@@ -124,7 +124,8 @@ def apply_associated_attributes!
       AssociatedGroup.where(id: associated_group_ids).update_all("last_used = CURRENT_TIMESTAMP")
     end
 
-    Group.user_trust_level_change!(user.id, user.trust_level) if user && !user.staged
+    # refreshes automatic group membership (despite the name, it doesn't change TLs)
+    Group.user_trust_level_change!(user.id, user.trust_level) if user
```

Comment:
Why was the `!user.staged` changed necessary?

Reply:
It was raised as a comment in https://github.com/discourse/discourse/pull/31759... it doesn't do anything because the user in this flow isn't staged anyway.